cql3/selection/abstract_function_selector.cc

```diff
@@ -98,6 +99,14 @@ abstract_function_selector::new_factory(shared_ptr<functions::function> fun, sha
         virtual bool is_aggregate_selector_factory() const override {
             return _fun->is_aggregate() || _factories->contains_only_aggregate_functions();
         }
+
+        virtual bool is_count_selector_factory() const override {
+            auto p = dynamic_cast<functions::abstract_function*>(_fun.get());
+            if (!p) {
+                return false;
+            }
+            return p->name().name == "countRows";
+        }
```

Comment:
As also mentioned in patch 6, I'd like to see this detected during prepare time instead of runtime.

Reply:
I did not see a response to this.

Is it possible to fool the check using `SELECT x as "countRows"`?

Reply:
I'm only judging by the code, but I think it won't be fooled - `AS X` is an alias, and the dynamic cast above specifically looks for an abstract_function instance. Unless it's possible to override the built-in "countRows" function with a non-aggregate or an aggregate that doesn't actually count rows, we should be safe.